tools/maint/create_entry_points.py

```diff
@@ -81,20 +85,22 @@ def generate_entry_points(cmd, path):
         bat_data = bat_data.replace('%~n0', entry_remap[entry_point].replace('/', '\\'))
         ps1_data = ps1_data.replace(r"$MyInvocation.MyCommand.Path -replace '\.ps1$', '.py'", fr'"$PSScriptRoot/{entry_remap[entry_point]}.py"')
 
-      out_sh_file = os.path.join(__rootdir__, entry_point)
-      with open(out_sh_file, 'w') as f:
-        f.write(sh_data)
-      os.chmod(out_sh_file, stat.S_IMODE(os.stat(out_sh_file).st_mode) | stat.S_IXUSR)
+      if do_unix:
+        out_sh_file = os.path.join(__rootdir__, entry_point)
+        with open(out_sh_file, 'w') as f:
+          f.write(sh_data)
+        os.chmod(out_sh_file, stat.S_IMODE(os.stat(out_sh_file).st_mode) | stat.S_IXUSR)
 
-      with open(os.path.join(__rootdir__, entry_point + '.bat'), 'w') as f:
-        f.write(bat_data)
+      if do_windows:
+        with open(os.path.join(__rootdir__, entry_point + '.bat'), 'w') as f:
+          f.write(bat_data)
 
-      with open(os.path.join(__rootdir__, entry_point + '.ps1'), 'w') as f:
-        f.write(ps1_data)
+        with open(os.path.join(__rootdir__, entry_point + '.ps1'), 'w') as f:
+          f.write(ps1_data)
 
   generate_entry_points(entry_points, os.path.join(__scriptdir__, 'run_python'))
   generate_entry_points(compiler_entry_points, os.path.join(__scriptdir__, 'run_python_compiler'))
 
 
 if __name__ == '__main__':
-  sys.exit(main())
+  sys.exit(main('--all' in sys.argv))
```

Comment:
Where is this going to be used from?

Reply:
Maybe nowhere?   I just figured some folks might want to generate all possible versions of the scripts.  As a developer its quite useful so you can see what would happen on a windows machine for example.